includes/class-subscription-lists.php

```diff
@@ -446,4 +450,48 @@ function( $list ) {
 			}
 		);
 	}
+
+	/**
+	 * Callback for the delete_post and wp_trash_post actions. Will remove the deleted/trashed list from the config.
+	 *
+	 * @param int           $post_id The Post ID.
+	 * @param false|WP_Post $post Informed by the delete_post action, but not by the wp_trash_post action. The deleted post object.
+	 * @return void
+	 */
+	public static function delete_post( $post_id, $post = false ) {
+		if ( ! $post ) {
+			$post = get_post( $post_id );
+		}
+		if ( ! $post instanceof WP_Post ) {
+			return;
+		}
+		if ( self::CPT !== $post->post_type ) {
+			return;
+		}
+		$list_config = Newspack_Newsletters_Subscription::get_lists_config();
+		if ( empty( $list_config ) || \is_wp_error( $list_config ) ) {
+			return;
+		}
+
+		$id = Subscription_List::FORM_ID_PREFIX . $post_id;
+		
+		if ( ! isset( $list_config[ $id ] ) ) {
+			return;
+		}
+		
+		unset( $list_config[ $id ] );
+
+		$new_list_config = [];
+		// generate a new list config without the deleted list.
+		foreach ( $list_config as $list_id => $list ) {
+			$new_list_config[] = [
+				'id'          => $list_id,
+				'active'      => $list['active'],
+				'title'       => $list['title'],
+				'description' => $list['description'],
+			];
+		}
```

Comment:
What's the purpose of regenerating the list after `unset()`?

Reply:
Now realizing, is it transforming from an array mapped by list IDs to a numbered array?

Reply:
NEVERMIND. That's how the `update_lists()` method receives params 😅

Reply:
yes... `get_lists_config` give me the list in one format but `update_lists` expect them in another format... Not very beatiful. We should avoid these loose parameters `$lists` which we don't really know what they are.